aws/data_source_aws_kms_alias.go

```diff
@@ -64,20 +64,24 @@ func dataSourceAwsKmsAliasRead(d *schema.ResourceData, meta interface{}) error {
 	d.SetId(time.Now().UTC().String())
 	d.Set("arn", alias.AliasArn)
 
-	aliasARN, err := arn.Parse(*alias.AliasArn)
-	if err != nil {
-		return err
-	}
-	targetKeyARN := arn.ARN{
-		Partition: aliasARN.Partition,
-		Service:   aliasARN.Service,
-		Region:    aliasARN.Region,
-		AccountID: aliasARN.AccountID,
-		Resource:  fmt.Sprintf("key/%s", *alias.TargetKeyId),
-	}
-	d.Set("target_key_arn", targetKeyARN.String())
+	// AWS service aliases do not return TargetKeyId:
```

Comment:
Nitpick: I would probably enhance this message so that we add the context `... do not return TargetKeyId when not associated with a Customer Managed Key (CMK)`. What do you think?

Reply:
Agree with @Ninir, there are cases where the key is associated, it is just not the case all the time

Reply:
I'll update this comment 👍